azure/spoke/variables.tf

```diff
@@ -85,32 +89,155 @@ variable "subnets" {
+    associate_default_network_security_group = optional(bool, true)
   }))
   default = {}
+}
+
+#########################
+# Network Security Group
+#########################
+
+variable "create_default_network_security_group" {
+  description = "Create a Network Security Group to associate with all subnets."
+  type        = bool
+  default     = true
+}
+
+variable "network_security_group_name" {
+  description = "Name of the default Network Security Group"
+  type        = string
+  default     = "default-nsg"
+}
+
+variable "nsg_rules_inbound" {
+  description = "A list of objects describing a rule inbound."
+  type = list(object({
+    rule = optional(string)
+    name = string
+    # nsg_name    = string
+    description = optional(string, "Created by Terraform.")
+
+    access   = optional(string, "Allow")
+    priority = optional(number)
+
+    protocol = optional(string, "*")
+    ports    = optional(set(string), ["*"])
+
+    source_prefixes      = optional(set(string), ["*"])
+    destination_prefixes = optional(set(string), ["VirtualNetwork"])
+
+    source_application_security_group_ids      = optional(set(string), null)
+    destination_application_security_group_ids = optional(set(string), null)
+  }))
+  default = []
+}
+
+variable "nsg_rules_outbound" {
+  description = "A list of objects describing a rule outbound."
+  type = list(object({
+    rule = optional(string)
+    name = string
+    # nsg_name    = string
+    description = optional(string, "Created by Terraform.")
+
+    access   = optional(string, "Allow")
+    priority = optional(number)
+
+    protocol = optional(string, "*")
+    ports    = optional(set(string), ["*"])
+
+    source_prefixes      = optional(set(string), ["*"])
+    destination_prefixes = optional(set(string), ["VirtualNetwork"])
+
+    source_application_security_group_ids      = optional(set(string), null)
+    destination_application_security_group_ids = optional(set(string), null)
+  }))
+  default = []
+}
+
+##############
+# Route Table
+##############
+
+variable "create_default_route_table" {
+  description = "Create a route table to associate with all subnets."
+  type        = bool
+  default     = true
+}
+
+variable "route_table_name" {
+  description = "Name of the default Route Table"
+  type        = string
+  default     = "default-rt"
+}
+
+
+variable "default_route_name" {
+  description = "Name of the default route."
+  type        = string
+  default     = "default-route"
+}
+
+variable "default_route_ip" {
+  description = "Default route IP Address."
+  type        = string
+
+  default = null
 
   validation {
-    error_message = "Must be valid IPv4 CIDR."
-    condition     = alltrue([for k, v in var.subnets : can(cidrhost(v.prefix, 0))])
+    condition     = can(regex("^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$", var.default_route_ip))
```

Comment:
Whilst this regex is valid, we can use cidrhost here instead as it's neater:

`can(cidrhost("${var.default_route_ip}/32", 0))`

Terraform Console EG:
```
> can(cidrhost("192.168.0.256/32",0))
false
> can(cidrhost("192.168.0.255/32",0))
true
```